package.json

```diff
@@ -162,8 +184,19 @@
               },
               "port": {
                 "type": "number",
-                "description": "Port to listen for remote debuggers.",
-                "default": 5000
+                "description": "Port to listen for remote debuggers."
+              },
+              "console": {
+                "type": "string",
+                "enum": [
+                  "deprecatedDebugConsole",
```

Comment:
That would certainly be my preference. I would then remove the two tests that fail with the new setup (syntax errors are no longer reported to the debugger). This would then also change the default behaviour for existing users when they upgrade the extension. Let me know and I'll update the PR.

Reply:
If the vs code are deprecating then it should be removed - the terminal would then take over right? (I have to checkout the code at give it a spin :) )

Reply:
Just to be clear, `console: 'deprecatedDebugConsole'` is the current mode of behaviour that mixes debugger and debuggee stdio when doing local debugging. I called it thus because it should be removed to address issue #58, and kept it in so that you have a chance to compare the modes directly, and in case you would want current users to be able to keep the current behaviour for a while. There are no upcoming changes in vscode that force us to remove it. But it seems we are on the same page here, and I'll update the PR to do away with this mode, I think that's the right choice.

Reply:
ok - it's the default configuration of perl db that makes perl db output in stderr.

In this case I think we should do some experiments in perl if the perl db can output in one stream stdout (like the remote version) - but I'm not sure if we can intercept the $DB:OUT to out put prefixed lines of data to stdout + intercept the application output to stdout aswell?

if we can have a stream output example:
```
#DB Perl debugger outputted something
#PR Hello world from the application
#DB Perl debugger output
etc...
```

Then we have the option to intercept both the debugger and application output. This would help solve the issue when debugging remotely and free up the stderr?

-

But the standard behaviour locally is to use two streams atm.

We could leave out the debugger output in the debug console or have that as an option - while adding a terminal output for the perl DB - let me know if I didn't explain myself well enough here :)

Reply:
Perhaps you could give the debug configuration `console: 'integratedTerminal'` as implemented here a try and tell me what you do not like about it from an end user perspective? I do not understand the angle from which you are looking at this at the moment.

Reply:
Sure, from the end user perspective: if the debug console can just be replaced by terminal window(s) that's fine - no need to complicate things

From my understanding running perl debugger in the terminal will make the extension run in `remote` debugging mode - we setup a session host and connect the client

My concern is that the remote debugging implementation can only receive either perl db output or the application output (or both? so one mixed stream)

For the client adapter to work it need to know if it's output from the debugger or the application - if you have ideas to solve this (or if it's not actually a problem) please share :)

Reply:
The `integratedTerminal`, `externalTerminal`, `remote`, and `none` modes talk exclusively to the debugger. They do not see it when the debuggee writes to its `stdout` or `stderr` and they do not interfere with the debuggee's `stdin`.

When a user wants to see what their application is printing, they do not use the debug console, but check in the terminal where the debuggee was launched (in the vscode terminal tab for integratedTerminal, in their terminal window outside vscode for externalTerminal, and on their remote terminal when they do remote debugging). In those terminals they can also directly interact with the debuggee, like when a script asks the user something interactively.

`deprecatedDebugConsole` is the »single process on local machine with mixed debugger/debuggee stdio« mode where it's hard to tell when the debugger is talking and when the application is talking. In this mode the user cannot interact with the application over `stdin`, the adapter misinterprets stuff printed to stderr as debugger output, ... hence `deprecated`.

Is that clearer?

Reply:
sorry I'm a bit slow on this - I'll have to checkout the branch and see it in action - if what you are saying works in general (local/remote) then that would be a brilliant solution

Reply:
@hoehrmann works great